### PR TITLE
Add commands for external Azure storage account configurations for we…

### PR DESCRIPTION
**Pull request by patricklee2**
eben's BYOS changes

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by patricklee2** [comment]
@yugangw-msft I tested the code against msftintch1501, its ready for review

**Comment by panchagnula** [comment]
@patricklee2  you need to add tests for the commands included in the PR.

File: src/command_modules/azure-cli-appservice/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 Release History
 ===============
+0.2.5
```
Comment (panchagnula) [inline src/command_modules/azure-cli-appservice/HISTORY.rst]: The SDK changes are not released yet so this line should be added to the 0.2.4 version

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 2]
```diff
@@ -119,6 +119,47 @@
         az webapp config appsettings set -g MyResourceGroup -n MyUniqueApp --settings WEBSITE_NODE_DEFAULT_VERSION=6.9.1
 """
 
+helps['webapp config azure-storage-accounts'] = """
```
Comment (panchagnula) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: what about function app?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: I would suggest `az webapp config storage-acount` as `storage account` pretty much means `azure storage account` . Generally, `azure` is a redundant term which I suggest remove from command/argument namings to get things simpler. Also avoid plurals in the command naming.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 3]
```diff
@@ -208,6 +208,14 @@ def load_arguments(self, _):
     with self.argument_context('webapp config connection-string') as c:
         c.argument('connection_string_type', options_list=['--connection-string-type', '-t'], help='connection string type', arg_type=get_enum_type(ConnectionStringType))
 
+    with self.argument_context('webapp config azure-storage-accounts') as c:
+        c.argument('id', options_list=['--id', '-i'], help='custom identifier')
+        c.argument('storage_type', options_list=['--storage-type', '-t'], help='storage type', arg_type=get_enum_type(AzureStorageType))
+        c.argument('account_name', options_list=['--acount-name', '-a'], help='storage account name')
+        c.argument('share_name', options_list=['--share-name', '-sn'], help='share name (Azure Files) or container name (Azure Blob Storage)')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: Short names with 2+ characters are not commended, please add one more dash, or just remove it.
Cc: @tjprescott

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py [hunk 4]
```diff
@@ -208,6 +208,14 @@ def load_arguments(self, _):
     with self.argument_context('webapp config connection-string') as c:
         c.argument('connection_string_type', options_list=['--connection-string-type', '-t'], help='connection string type', arg_type=get_enum_type(ConnectionStringType))
 
+    with self.argument_context('webapp config azure-storage-accounts') as c:
+        c.argument('id', options_list=['--id', '-i'], help='custom identifier')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py]: Please use `--custom-id`, as `id` is a term mostly reserved for resource ids

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/commands.py [hunk 5]
```diff
@@ -107,6 +107,12 @@ def load_command_table(self, _):
         g.custom_command('set', 'update_connection_strings')
         g.custom_command('delete', 'delete_connection_strings')
 
+    with self.command_group('webapp config azure-storage-accounts') as g:
+        g.custom_command('list', 'get_azure_storage_accounts', exception_handler=empty_on_404)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/commands.py]: Can you double check whether or not `empty_on_404` is needed? If service end doesn't throw, then please remove it.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/recordings/test_webapp_config.yaml [hunk 6]
```diff
@@ -599,416 +599,92 @@ interactions:
...
-    body: {string: '{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_webapp_config000001/providers/Microsoft.Web/sites/webapp-config-test000002/config/connectionstrings","name":"connectionstrings","type":"Microsoft.Web/sites/config","location":"West
-        US","properties":{"c2":{"value":"conn2","type":"MySql"}}}'}
-    headers:
-      cache-control: [no-cache]
-      content-length: ['384']
-      content-type: [application/json]
-      date: ['Thu, 06 Sep 2018 14:54:52 GMT']
+    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_webapp_config000001/providers/Microsoft.Web/sites/webapp-config-test000002/config/azurestorageaccounts/list?api-version=2018-02-01
+  response:
+    body: {string: '{"Code":"MethodNotAllowed","Message":"<?xml version=\"1.0\" encoding=\"utf-8\"?>\r\n<!DOCTYPE
```
Comment (patricklee2) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/recordings/test_webapp_config.yaml]: this wont pass until the api is fixed in ant 77

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py (hunk 2 above)
Comment (patricklee2) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: the functionapp runtime doesn't use this, so we dont need to add it

Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: @patricklee2 , i am guessing you mean the feature is not supported on function app as yet. If so yes, don't need to add support here. Please do update the command name as per Yugang's comments.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/commands.py (hunk 5 above)
Comment (patricklee2) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/commands.py]: i think we need it

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 7]
```diff
@@ -320,6 +320,37 @@ def test_webapp_config(self, resource_group):
                      JMESPathCheck('length(@)', 1),
                      JMESPathCheck('[0].name', '{0}.azurewebsites.net'.format(webapp_name))])
 
+        # site azure storage account configurations tests
+        # add
+        byos_resource_group = 'yili-cus-stage-01' # on the beta stamp msftintch1501
```
Comment (patricklee2) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py]: points this test to the beta stamp

**Pull request merged**